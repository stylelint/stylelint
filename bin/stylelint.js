#!/usr/bin/env node

'use strict';

// to use V8's code cache to speed up instantiation time
require('v8-compile-cache');

require('../lib/cli')(process.argv.slice(2));
