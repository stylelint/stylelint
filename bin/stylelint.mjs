#!/usr/bin/env node

import cli from '../lib/cli.mjs';

cli(process.argv.slice(2));
