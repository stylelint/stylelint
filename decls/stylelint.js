export type stylelint$configExtends = string | Array<string>;
export type stylelint$configPlugins = string | Array<string>;
export type stylelint$configProcessors =
  | string
  | Array<string | [string, Object]>;
export type stylelint$configIgnoreFiles = string | Array<string>;

export type stylelint$configRuleSettings = any | [any, Object];
export type stylelint$configRules = {
  [ruleName: string]: stylelint$configRuleSettings
};

export type stylelint$config = {
  extends?: stylelint$configExtends,
  plugins?: stylelint$configPlugins,
  pluginFunctions?: {
    [pluginName: string]: Function
  },
  processors?: stylelint$configProcessors,
  processorFunctions?: Array<Function>,
  ignoreFiles?: stylelint$configIgnoreFiles,
  ignorePatterns?: string,
  rules?: stylelint$configRules,
  codeProcessors?: Array<Function>,
  resultProcessors?: Array<Function>,
  quiet?: boolean
};

export type stylelint$syntaxes = "scss" | "less" | "sugarss";

export type stylelint$options = {
  config?: stylelint$config,
  configFile?: string,
  configBasedir?: string,
  configOverrides?: Object,
  ignoreDisables?: boolean,
  ignorePath?: string,
  reportNeedlessDisables?: boolean,
  syntax?: stylelint$syntaxes,
  customSyntax?: string,
  fix?: boolean
};

export type stylelint$internalApi = {
  _options: stylelint$options,
  _extendExplorer: { load: Function },
  _fullExplorer: { load: Function },
  _configCache: Map<string, Object>,
  _specifiedConfigCache: Map<string, Object>,
  _postcssResultCache: Map<string, Object>,

  _augmentConfig: Function,
  _getPostcssResult: Function,
  _lintSource: Function,
  _createStylelintResult: Function,
  _createEmptyPostcssResult?: Function,

  getConfigForFile: Function,
  isPathIgnored: Function,
  lintSource: Function
};

export type stylelint$warning = {
  line: number,
  column: number,
  rule: string,
  severity: string,
  text: string
};

export type stylelint$result = {
  source: string,
  deprecations: Array<{
    text: string,
    reference: string
  }>,
  invalidOptionWarnings: Array<{
    text: string
  }>,
  parseErrors: Array<stylelint$warning>,
  errored?: boolean,
  warnings: Array<stylelint$warning>,
  ignored?: boolean,
  _postcssResult?: Object
};

export type stylelint$needlessDisablesReport = Array<{
  source: string,
  ranges: Array<{
    source: string,
    ranges: Array<{
      start: number,
      end?: number
    }>
  }>
}>;

export type stylelint$standaloneReturnValue = {
  results: Array<stylelint$result>,
  errored: boolean,
  output: any,
  needlessDisables?: stylelint$needlessDisablesReport
};

export type stylelint$standaloneOptions = {
  files?: string | Array<string>,
  cache?: boolean,
  cacheLocation?: string,
  code?: string,
  codeFilename?: string,
  config?: stylelint$config,
  configFile?: string,
  configBasedir?: string,
  configOverrides?: Object,
  ignoreDisables?: boolean,
  ignorePath?: string,
  reportNeedlessDisables?: boolean,
  syntax?: stylelint$syntaxes,
  customSyntax?: string,
  formatter?: "json" | "string" | "verbose" | Function,
  disableDefaultIgnores?: boolean,
  allowEmptyInput?: boolean,
  fix?: boolean
};
