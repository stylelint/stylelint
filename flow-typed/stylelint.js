export type stylelint$configExtends = string | Array<string>;
export type stylelint$configPlugins = string | Array<string>;
export type stylelint$configProcessors =
  | string
  | Array<string | [string, Object]>;
export type stylelint$configIgnoreFiles = string | Array<string>;

export type stylelint$configRuleSettings = any | [any, Object];
export type stylelint$configRules = {
  [ruleName: string]: stylelint$configRuleSettings
};

export type stylelint$config = {
  extends?: stylelint$configExtends,
  plugins?: stylelint$configPlugins,
  pluginFunctions?: {
    [pluginName: string]: Function
  },
  processors?: stylelint$configProcessors,
  processorFunctions?: Array<Function>,
  ignoreFiles?: stylelint$configIgnoreFiles,
  ignorePatterns?: string,
  rules?: stylelint$configRules,
  codeProcessors?: Array<Function>,
  resultProcessors?: Array<Function>,
  quiet?: boolean,
  defaultSeverity?: string
};

export type stylelint$syntaxes = "scss" | "less" | "sugarss";

export type stylelint$options = {
  config?: stylelint$config,
  configFile?: string,
  configBasedir?: string,
  configOverrides?: Object,
  ignoreDisables?: boolean,
  ignorePath?: string,
  reportInvalidScopeDisables?: boolean,
  reportNeedlessDisables?: boolean,
  syntax?: stylelint$syntaxes,
  customSyntax?: string,
  fix?: boolean
};

export type stylelint$internalApi = {
  _options: stylelint$options,
  _extendExplorer: {
    search: string => Promise<null | Object>,
    load: string => Promise<null | Object>
  },
  _fullExplorer: {
    search: string => Promise<null | Object>,
    load: string => Promise<null | Object>
  },
  _configCache: Map<string, Object>,
  _specifiedConfigCache: Map<string, Object>,
  _postcssResultCache: Map<string, Object>,

  _augmentConfig: Function,
  _getPostcssResult: Function,
  _lintSource: Function,
  _createStylelintResult: Function,
  _createEmptyPostcssResult?: Function,

  getConfigForFile: Function,
  isPathIgnored: Function,
  lintSource: Function
};

export type stylelint$warning = {
  line: number,
  column: number,
  rule: string,
  severity: string,
  text: string
};

export type stylelint$result = {
  source: string,
  deprecations: Array<{
    text: string,
    reference: string
  }>,
  invalidOptionWarnings: Array<{
    text: string
  }>,
  parseErrors: Array<stylelint$warning>,
  errored?: boolean,
  warnings: Array<stylelint$warning>,
  ignored?: boolean,
  _postcssResult?: Object
};

export type stylelint$cssSyntaxError = {
  column: number,
  file?: string,
  input: {
    column: number,
    file?: string,
    line: number,
    source: string
  },
  line: number,
  message: string,
  name: string,
  reason: string,
  source: string
};

export type stylelint$disableOptionsReport = Array<{
  source: string,
  ranges: Array<{
    unusedRule: string,
    start: number,
    end?: number
  }>
}>;

export type stylelint$standaloneReturnValue = {
  results: Array<stylelint$result>,
  errored: boolean,
  output: any,
  maxWarningsExceeded?: {
    maxWarnings: number,
    foundWarnings: number
  },
  needlessDisables?: stylelint$disableOptionsReport,
  invalidScopeDisables?: stylelint$disableOptionsReport
};

export type stylelint$standaloneOptions = {
  files?: string | Array<string>,
  globbyOptions?: Object,
  cache?: boolean,
  cacheLocation?: string,
  code?: string,
  codeFilename?: string,
  config?: stylelint$config,
  configFile?: string,
  configBasedir?: string,
  configOverrides?: Object,
  printConfig?: string,
  ignoreDisables?: boolean,
  ignorePath?: string,
  ignorePattern?: RegExp,
  reportNeedlessDisables?: boolean,
  reportInvalidScopeDisables?: boolean,
  maxWarnings?: number,
  syntax?: stylelint$syntaxes,
  customSyntax?: string,
  formatter?: "compact" | "json" | "string" | "unix" | "verbose" | Function,
  disableDefaultIgnores?: boolean,
  fix?: boolean,
  allowEmptyInput: boolean
};
