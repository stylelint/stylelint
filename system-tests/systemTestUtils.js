"use strict";

const _ = require("lodash");
const path = require("path");
const stripAnsi = require("strip-ansi");

function caseFilePath(caseNumber, fileName) {
  return path.join(__dirname, caseNumber, fileName);
}

function caseStylesheetGlob(caseNumber) {
  return caseFilePath(caseNumber, "stylesheet.*");
}

function caseConfig(caseNumber, ext) {
  ext = ext || "json";

  return caseFilePath(caseNumber, `config.${ext}`);
}

function prepResults(results) {
  return results.map(result => {
    // The _postcssResult object is not part of our API and is huge
    const preppedResult = _.omit(result, "_postcssResult");

    // The `source` of each file will not be the same on different machines or platforms
    preppedResult.source = path
      .relative(__dirname, result.source)
      .replace(new RegExp(`\\${path.sep}`, "g"), "/");

    return preppedResult;
  });
}

function stripColors(input) {
  return stripAnsi(input);
}

module.exports = {
  caseFilePath,
  caseStylesheetGlob,
  caseConfig,
  prepResults,
  stripColors
};
