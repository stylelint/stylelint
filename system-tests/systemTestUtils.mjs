import { copyFile, readFile } from 'node:fs/promises';
import { fileURLToPath } from 'node:url';
import os from 'node:os';
import path from 'node:path';
import process from 'node:process';

import replaceBackslashes from '../lib/testUtils/replaceBackslashes.mjs';
import uniqueId from '../lib/testUtils/uniqueId.mjs';

const __dirname = fileURLToPath(new URL('.', import.meta.url));

export function caseFilePath(caseNumber, fileName = 'stylesheet') {
	return replaceBackslashes(path.join(__dirname, caseNumber, fileName));
}

export function caseFiles(caseNumber) {
	return caseFilePath(caseNumber, 'stylesheet.*');
}

export function caseConfigFile(caseNumber, ext = 'json') {
	return caseFilePath(caseNumber, `config.${ext}`);
}

export async function caseConfig(caseNumber) {
	return JSON.parse(await readFile(caseConfigFile(caseNumber, 'json'), 'utf8'));
}

export async function caseCode(caseNumber, ext = 'css') {
	return readFile(`${caseFilePath(caseNumber)}.${ext}`, 'utf8');
}

export async function caseFilesForFix(caseNumber, ext = 'css') {
	const tempPath = replaceBackslashes(path.join(os.tmpdir(), `stylesheet-${uniqueId()}.${ext}`));

	await copyFile(path.join(__dirname, caseNumber, `stylesheet.${ext}`), tempPath);

	return tempPath;
}

export function prepForSnapshot({ results, cwd, output, report, ...rest }) {
	// If output isn't fixed code
	if (output.startsWith('[')) {
		// The `source` of each file varies between platforms or if a tmp file is used
		output = JSON.parse(output).map((warning) => ({
			...warning,
			source: '/path/to/dummy.css',
		}));
	}

	if (report) {
		// The `source` of each file varies between platforms or if a tmp file is used
		report = JSON.parse(report).map((warning) => ({
			...warning,
			source: '/path/to/dummy.css',
		}));
	}

	return {
		cwd: path.relative(process.cwd(), cwd),
		// The _postcssResult object is not part of our API and is huge
		results: results.map((result) => {
			delete result._postcssResult;

			return { ...result, source: '/path/to/dummy.css' };
		}),
		output, // TODO: Deprecated. Remove in the next major version.
		report,
		...rest,
	};
}
