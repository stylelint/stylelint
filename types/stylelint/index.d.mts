export {
	default,
	BuiltInRules,
	Config,
	ConfigRuleSettings,
	CosmiconfigResult,
	CssSyntaxError,
	CustomSyntax,
	DisableOptions,
	DisableOptionsReport,
	DisablePropertyName,
	DisableReportRange,
	DisabledRange,
	DisabledRangeObject,
	DisabledWarning,
	Formatter,
	FormatterType,
	Formatters,
	GetLintSourceOptions,
	GetPostcssOptions,
	InternalApi,
	LintResult,
	LinterOptions,
	LinterResult,
	LonghandSubPropertiesOfShorthandProperties,
	Plugin,
	PostcssPluginOptions,
	PostcssResult,
	Problem,
	PublicApi,
	Rule,
	RuleBase,
	RuleContext,
	RuleMessage,
	RuleMessages,
	RuleMeta,
	RuleOptions,
	RuleOptionsPossible,
	RuleSeverity,
	Severity,
	ShorthandProperties,
	StylelintPostcssResult,
	Utils,
	Warning,
	WarningOptions,
} from './index.js';
