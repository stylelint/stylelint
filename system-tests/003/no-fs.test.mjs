import stylelint from '../../lib/index.mjs';

import { caseCode, caseConfig, prepForSnapshot } from '../systemTestUtils.mjs';

const CASE_NUMBER = '003';

it('no-fs - zen garden CSS with standard config', async () => {
	expect(
		prepForSnapshot(
			await stylelint.lint({
				code: await caseCode(CASE_NUMBER),
				config: await caseConfig(CASE_NUMBER),
				fix: true,
			}),
		),
	).toMatchSnapshot();
}, 10000);
