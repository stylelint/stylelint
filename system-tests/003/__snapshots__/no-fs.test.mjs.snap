// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-fs - zen garden CSS with standard config 1`] = `
{
  "cwd": "",
  "errored": true,
  "output": "/* css Zen Garden default style v1.02 */

/* css released under Creative Commons License - http://creativecommons.org/licenses/by-nc-sa/1.0/  */

/* This file based on 'Tranquille' by Dave Shea */

/* You may use this file as a foundation for any new work, but you may find it easier to start from scratch. */

/* Not all elements are defined in this file, so you'll most likely want to refer to the xhtml as well. */

/* Your images should be linked as if the CSS file sits in the same folder as the images. ie. no paths. */


/* basic elements */
html {
	margin: 0;
	padding: 0;
	}

body {
	font: 75% georgia, sans-serif;
	line-height: 1.88889;
	color: #555753;
	background: #fff url(http://csszengarden.com/001/blossoms.jpg) no-repeat bottom right;
	margin: 0;
	padding: 0;
	}

p {
	margin-top: 0;
	text-align: justify;
	}

h3 {
	font: italic normal 1.4em georgia, sans-serif;
	letter-spacing: 1px;
	margin-bottom: 0;
	color: #7D775C;
	}

a:link {
	font-weight: bold;
	text-decoration: none;
	color: #B7A5DF;
	}

a:visited {
	font-weight: bold;
	text-decoration: none;
	color: #D4CDDC;
	}

a:hover, a:focus, a:active {
	text-decoration: underline;
	color: #9685BA;
	}

abbr {
	border-bottom: none;
	}


/* specific divs */
.page-wrapper {
	background: url(http://csszengarden.com/001/zen-bg.jpg) no-repeat top left;
	padding: 0 175px 0 110px;
	margin: 0;
	position: relative;
	}

.intro {
	min-width: 470px;
	width: 100%;
	}

header h1 {
	background: transparent url(http://csszengarden.com/001/h1.gif) no-repeat top left;
	margin-top: 10px;
	display: block;
	width: 219px;
	height: 87px;
	float: left;
	text-indent: 100%;
	white-space: nowrap;
	overflow: hidden;
	}

header h2 {
	background: transparent url(http://csszengarden.com/001/h2.gif) no-repeat top left;
	margin-top: 58px;
	margin-bottom: 40px;
	width: 200px;
	height: 18px;
	float: right;
	text-indent: 100%;
	white-space: nowrap;
	overflow: hidden;
	}

header {
	padding-top: 20px;
	height: 87px;
}

.summary {
	clear: both;
	margin: 20px 20px 20px 10px;
	width: 160px;
	float: left;
	}

.summary p {
	font: italic 1.1em/2.2 georgia;
	text-align: center;
	}

.preamble {
	clear: right;
	padding: 0 10px 0;
	}

.supporting {
	padding-left: 10px;
	margin-bottom: 40px;
	}

footer {
	text-align: center;
	}

footer a:link, footer a:visited {
	margin-right: 20px;
	}

.sidebar {
	margin-left: 600px;
	position: absolute;
	top: 0;
	right: 0;
	}

.sidebar .wrapper {
	font: 10px verdana, sans-serif;
	background: transparent url(http://csszengarden.com/001/paper-bg.jpg) top left repeat-y;
	padding: 10px;
	margin-top: 150px;
	width: 130px;
	}

.sidebar h3.select {
	background: transparent url(http://csszengarden.com/001/h3.gif) no-repeat top left;
	margin: 10px 0 5px;
	width: 97px;
	height: 16px;
	text-indent: 100%;
	white-space: nowrap;
	overflow: hidden;
	}

.sidebar h3.archives {
	background: transparent url(http://csszengarden.com/001/h5.gif) no-repeat top left;
	margin: 25px 0 5px;
	width:57px;
	height: 14px;
	text-indent: 100%;
	white-space: nowrap;
	overflow: hidden;
	}

.sidebar h3.resources {
	background: transparent url(http://csszengarden.com/001/h6.gif) no-repeat top left;
	margin: 25px 0 5px;
	width:63px;
	height: 10px;
	text-indent: 100%;
	white-space: nowrap;
	overflow: hidden;
	}


.sidebar ul {
	margin: 0;
	padding: 0;
	}

.sidebar li {
	line-height: 1.3em;
	background: transparent url(http://csszengarden.com/001/cr1.gif) no-repeat top center;
	display: block;
	padding-top: 5px;
	margin-bottom: 5px;
	list-style-type: none;
	}

.sidebar li a:link {
	color: #988F5E;
	}

.sidebar li a:visited {
	color: #B3AE94;
	}


.extra1 {
	background: transparent url(http://csszengarden.com/001/cr2.gif) top left no-repeat;
	position: absolute;
	top: 40px;
	right: 0;
	width: 148px;
	height: 110px;
	}
",
  "reportedDisables": [],
  "results": [
    {
      "deprecations": [],
      "errored": true,
      "ignored": undefined,
      "invalidOptionWarnings": [],
      "parseErrors": [],
      "warnings": [
        {
          "column": 25,
          "endColumn": 32,
          "endLine": 102,
          "line": 102,
          "rule": "font-family-no-missing-generic-family-keyword",
          "severity": "error",
          "text": "Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)",
        },
      ],
    },
  ],
  "ruleMetadata": {
    "at-rule-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/at-rule-empty-line-before",
    },
    "at-rule-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/at-rule-no-unknown",
    },
    "block-no-empty": {
      "url": "https://stylelint.io/user-guide/rules/block-no-empty",
    },
    "color-hex-length": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/color-hex-length",
    },
    "color-no-invalid-hex": {
      "url": "https://stylelint.io/user-guide/rules/color-no-invalid-hex",
    },
    "comment-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/comment-empty-line-before",
    },
    "comment-no-empty": {
      "url": "https://stylelint.io/user-guide/rules/comment-no-empty",
    },
    "comment-whitespace-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/comment-whitespace-inside",
    },
    "custom-property-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/custom-property-empty-line-before",
    },
    "declaration-block-no-duplicate-properties": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-no-duplicate-properties",
    },
    "declaration-block-no-shorthand-property-overrides": {
      "url": "https://stylelint.io/user-guide/rules/declaration-block-no-shorthand-property-overrides",
    },
    "declaration-block-single-line-max-declarations": {
      "url": "https://stylelint.io/user-guide/rules/declaration-block-single-line-max-declarations",
    },
    "declaration-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-empty-line-before",
    },
    "font-family-no-duplicate-names": {
      "url": "https://stylelint.io/user-guide/rules/font-family-no-duplicate-names",
    },
    "font-family-no-missing-generic-family-keyword": {
      "url": "https://stylelint.io/user-guide/rules/font-family-no-missing-generic-family-keyword",
    },
    "function-calc-no-unspaced-operator": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-calc-no-unspaced-operator",
    },
    "function-linear-gradient-no-nonstandard-direction": {
      "url": "https://stylelint.io/user-guide/rules/function-linear-gradient-no-nonstandard-direction",
    },
    "function-name-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-name-case",
    },
    "keyframe-declaration-no-important": {
      "url": "https://stylelint.io/user-guide/rules/keyframe-declaration-no-important",
    },
    "length-zero-no-unit": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/length-zero-no-unit",
    },
    "media-feature-name-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/media-feature-name-no-unknown",
    },
    "no-descending-specificity": {
      "url": "https://stylelint.io/user-guide/rules/no-descending-specificity",
    },
    "no-duplicate-at-import-rules": {
      "url": "https://stylelint.io/user-guide/rules/no-duplicate-at-import-rules",
    },
    "no-duplicate-selectors": {
      "url": "https://stylelint.io/user-guide/rules/no-duplicate-selectors",
    },
    "no-empty-source": {
      "url": "https://stylelint.io/user-guide/rules/no-empty-source",
    },
    "no-invalid-double-slash-comments": {
      "url": "https://stylelint.io/user-guide/rules/no-invalid-double-slash-comments",
    },
    "property-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/property-no-unknown",
    },
    "rule-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/rule-empty-line-before",
    },
    "selector-pseudo-class-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown",
    },
    "selector-pseudo-element-colon-notation": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-element-colon-notation",
    },
    "selector-pseudo-element-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-element-no-unknown",
    },
    "selector-type-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-type-case",
    },
    "selector-type-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-type-no-unknown",
    },
    "string-no-newline": {
      "url": "https://stylelint.io/user-guide/rules/string-no-newline",
    },
    "unit-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/unit-no-unknown",
    },
    "value-keyword-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-keyword-case",
    },
  },
}
`;
